pipeline/state_duration.go

```diff
@@ -0,0 +1,58 @@
+package pipeline
+
+import (
+	"time"
+
+	"github.com/influxdata/kapacitor/tick/ast"
+)
+
+// Compute the duration of a given state.
+// The state is defined via a lambda expression. For each consecutive point for
+// which the expression evaluates as true, the state duration will be
+// incremented by the duration between points. When a point evaluates as false,
+// the state duration is reset to 0.
+//
+// The state duration will be added as an additional field to each point. If the
+// expression evaluates as false, the field will not be added. If the expression
```

Comment:
Sounds good, lets default the value to -1 like you suggest. That way a value of >= 0 indicates that the state condition is true and can always be interpreted as the duration of the state. By using -1 as the default a value is always set so a `|default` node is not required and then it can easily be changed on via an eval node.

Thanks for working through this with me.